src/lib.rs

```diff
@@ -194,6 +200,48 @@ fn handle_error(app: &Config, req: &mut Request, err: BrokerError) -> IronResult
 }
 
 
+/// Middleware that logs each request.
+pub struct LogMiddleware;
+impl BeforeMiddleware for LogMiddleware {
+    fn before(&self, req: &mut Request) -> IronResult<()> {
+        info!("{} {}", req.method, req.url);
+        Ok(())
+    }
+}
+
+
+/// Middleware that sets common headers.
+pub struct DefaultHeadersMiddleware;
+impl DefaultHeadersMiddleware {
+    fn set_headers(&self, res: &mut Response) {
+        // Specify a tight content security policy. We need to be able to POST
+        // redirect anywhere, and run our own inline scripts.
+        let csp = vec![
+            "sandbox allow-scripts allow-forms",
+            "default-src 'none'",
+            "script-src 'unsafe-inline'",
+            "form-action *",
+        ].join("; ");
+        res.set_mut((modifiers::Header(StrictTransportSecurity::including_subdomains(31536000u64)),
```

Comment:
Let's send a plain HSTS header; I'm not confident we're ready for `includeSubdomains` yet, since my understanding is that it would apply to *all* domains of `portier.io`, or wherever the broker is running. And then have year-long consequences for a botched roll-out.

This could be especially dangerous for folks running their own brokers inadvertently locking themselves out of other, non-HTTPS subdomains on an origin.

For us, neither `https://portier.io` nor `https://demo.portier.io` are available, both redirect to another location *with* HTTPS.

With that changed, I'm +1 on merging.